forte/data/ontology/ontology_code_generator.py

```diff
@@ -0,0 +1,491 @@
+"""
+    Module to automatically generate python ontology given json file
+    Performs a preliminary check of dependencies
+"""
+import os
+import json
+import logging
+import tempfile
+from pathlib import Path
+from collections import defaultdict
+from distutils import dir_util
+
+from types import ModuleType
+from typing import Dict, List, Optional, Tuple, Set
+
+import typed_ast.ast3 as ast
+import typed_astunparse as ast_unparse
+
+from forte.data.ontology import utils, top
+from forte.data.ontology.code_generation_util import (
+    BasicItem, CompositeItem, DefinitionItem, FileItem)
+
+logging.basicConfig(level=os.environ.get("LOGLEVEL", "INFO"))
+log = logging.getLogger(__name__)
+
+
+class OntologyCodeGenerator:
+    """
+    Class to generate python ontology given ontology config in json format
+    Salient Features -
+        (1) Generates a class for each entry in the module corresponding to
+        the defined entry package
+        (2) Dependencies to other json configs through the json `imports`
+    Example:
+        >>> destination_dir = OntologyCodeGenerator().generate_ontology(
+        'test/example_ontology_config.json')
+    """
+    AUTO_GEN_CONST = '***automatically_generated***'
+
+    def __init__(self,
+                 json_dir_paths: Optional[List[str]] = None,
+                 top_level_dir: Optional[str] = None,
+                 ontology_base_module: ModuleType = top):
+        """
+        Args:
+            ontology_base_module: The entries of the `base_ontology_module`
+            serve as ancestors of the user-defined entries.
+            json_dir_paths: Additional user provided paths to search the
+            imported json configs from. By default paths provided in the json
+            configs and the current working directory would be searched.
+            indent: Number of indent required in the indentation.
+        """
```

Comment:
The docstring does not match the function signature, and probably the indent of the docstring is not correct.